lang/csharp/src/apache/test/Avro.test.csproj

```diff
@@ -17,8 +17,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks Condition="'$(OS)'!='Windows_NT'">netcoreapp3.1</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)'=='Windows_NT'">net461;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)'!='Windows_NT'">net5.0</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)'=='Windows_NT'">net461;net5.0</TargetFrameworks>
```

Comment:
> I compiled the following table of updated dependencies in this PR. We should probably list these out in the commit message and/or Jira issue.
> 
> Package	Old	New
> Microsoft.CodeAnalysis.FxCopAnalyzers	2.9.8	3.3.1
> Newtonsoft.Json	10.0.3	12.0.3
> System.CodeDom	4.4.0	5.0.0
> System.Reflection.Emit.ILGeneration	4.3.0	4.7.0
> System.Reflection.Emit.Lightweight	4.3.0	4.7.0
> Microsoft.Build.Framework	15.6.82	16.8.0
> Microsoft.Build.Utilities.Core	15.6.82	16.8.0
> nunit3testadapter	3.16.1	3.17.0
> NUnit.ConsoleRunner	3.10.0	3.11.1
> Microsoft.NET.Test.Sdk	16.4.0	16.8.0

Usually what I do in .net projects, I add a Versions.prtops file int h eroot and all csproj files include it, just like the Common.props. This file only have PACKAGENAMEVersion defines in them. e.g. "NewtonsoftJsonVersion" = "12.0.3". Every csproj in the source tree will use only PACKAGENAMEVersion defines in the PackageReference, so the version "upgrade" is clean and simple, only change in one place.

Reply:
> It may be good for us to continue testing against .NET Core 3.1, just in case there was some compatibility issue between the two.

Good catch. Now thinking about this. I think it should be netcoreapp2.1 added to it (or 2.1 and 3.1). So here is my logic, let me know if I miss or misunderstand something.

Avro.main is compiled for netstandard2.0, netstandard2.1 and netcoreapp2.1. Netstandard2.1 and netcoreapp2.1 (which is btw is very much like netstandard2.1, without having netstandard2.1 support). There is no other framework in Avro.main, since MS recommends to add netcoreapp or other frameweorks only if the library is really compiled with some specific framweork feature/API (using ifdef). SO we are good there.

When Avro.test is tested, all frameworks will be tested in the TargetFramworks list (if no --framework is specified). 
netf461 will use netstandard2.0 so netstandard2.0 version of Avro.main is tested.
netcoreapp2.1 fw of Avro.test would be linked against the netcoreapp2.1 version of Avro.main since that specific fw exists in the nuget package and Avro.main.
netcoreapp3.1 or net5.0 will link against netstandard2.1, since both of them support it, and msbuild will be pick the most appropiate and (best) choice of matching fw.

Honsetly, since the test is very fast, I would add all supported frtameworks to Avro.test. net461,netcoreapp2.1, netcoreepp3.1 and net5.0 (it might test netstandard2.1 2x.).

Reply:
Wow, that's great! Yes, I  agree that we should just run all the supported frameworks in Avro.test (net461, netcoreapp2.1, netcoreapp3.1 and net5.0) for the sake of simplicity and since the tests are so quick.